xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp

```diff
@@ -618,6 +618,16 @@ bool CAESinkALSA::Initialize(AEAudioFormat &format, std::string &device)
   }
   // adjust format to the configuration we got
   format.m_channelLayout = GetChannelLayout(format, outconfig.channels);
+  // we might end up with an unusable channel layout starts with UNKNOWN
+  // channel, let's do a sanity check. UNKNOWN channels are only allowed for
+  // padding.
+  bool isSaneLayout = true;
+  if ((format.m_channelLayout.Count() > 0) && format.m_channelLayout.IsChannelUnknown(0));
```

Comment:
testing for IsChannelValid would seem more reasonalbe to me.
it should not only test index 0 but against the spec. FL, FL, UNKNOWN, BL, BR ist not a valid map